remoteworkitem/flatten.go

```diff
@@ -0,0 +1,43 @@
+package remoteworkitem
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Flatten Takes the nested map and returns a non nested one with dot delimited keys
+func Flatten(source map[string]interface{}) map[string]interface{} {
+	target := make(map[string]interface{})
+
+	flatten(target, source, nil)
+
+	return target
+}
+
+func flatten(target map[string]interface{}, source map[string]interface{}, parent *string) {
+	for k, v := range source {
+		var key string
+		if parent == nil {
+			key = k
+		} else {
+			key = *parent + "." + k
+		}
+
+		if v != nil && reflect.TypeOf(v).Kind() == reflect.Map {
+			flatten(target, v.(map[string]interface{}), &key)
```

Comment:
We are not checking the key type (string) here...this may crash for unexpected input. Generally, I think a general function like this should have an error return for unexpected input.